### PR TITLE
Ops 441 initial page and combobox

### PR DESCRIPTION
**Pull request by johndeange**
## What changed

- Added `User Admin` page (only accessible by users with the `USER_ADMIN` role)
- Added `Budget Team` user with roles `BUDGET_TEAM` and `USER_ADMIN`
- Updated `UserInfo` component to make editable by `USER_ADMIN` (and display human readable values)
- Added new RTKQ API `opsAuthAPI`
- Improvements/fixes to `msw` JS stub (`mocks.js`)

## Issue

https://github.com/HHS/OPRE-OPS/issues/441

## How to test

1. Testing that only `USER_ADMIN` role can access the `User Admin` menu option

- Login as `Admin` and the menu option should not be visible.
- Login as `Budget Team` and the option should be visible (since BT is a `USER_ADMIN`)

2. Testing that only `USER_ADMIN` has an editable view of `UserInfo`

- Login as `Admin` (or any user other than `Budget Team`)
- Click on the user's email address in the upper right of the page to go to the `User Details` page.
- Verify that the `UserInfo` is read-only.

- Login as `Budget Team`
- Click on the user's email address in the upper right of the page to go to the `User Details` page.
- Verify that the `UserInfo` is editable, i.e. has combo boxes for status, role, and division

N.B. The actual persistence of the `UserInfo` changes will be covered in a future PR.   (There is already too much going on in this PR)

## Screenshots

_If relevant, e.g. for a front-end feature_

## Definition of Done Checklist
- [x] OESA: Code refactored for clarity
- [x] OESA: Dependency rules followed
- [x] Automated unit tests updated and passed
- [x] Automated integration tests updated and passed
- [x] Automated quality tests updated and passed
- [x] Automated load tests updated and passed
- [x] Automated a11y tests updated and passed
- [x] Automated security tests updated and passed
- [x] 90%+ Code coverage achieved
- [x] Form validations updated

### CODE REVIEW

**Comment by fpigeonjr** [comment]
👀
image

**Comment by fpigeonjr** [comment]
> Login as Division Director and the option should be visible (since DD is a USER_ADMIN)

When logged in as as Director Dave, I am not seeing the **User Admin** tab

SCR-20240807-plho

**Comment by fpigeonjr** [comment]
🥇 UserInfo is only editable by the Budget Team 

image

File: frontend/src/components/Users/UserInfo/UserInfo.jsx [hunk 1]
```diff
@@ -1,55 +1,119 @@
...
 
-const UserInfo = () => {
-    const user = useSelector((state) => state.userDetail.user);
+const UserInfo = ({ user, isEditable }) => {
+    const [selectedDivision, setSelectedDivision] = React.useState({});
+    const [selectedStatus, setSelectedStatus] = React.useState({});
+    const [selectedRoles, setSelectedRoles] = React.useState([]);
+    const statusData = [
+        { id: 1, name: "ACTIVE" },
+        { id: 2, name: "INACTIVE" },
```
Comment (fpigeonjr) [inline frontend/src/components/Users/UserInfo/UserInfo.jsx:13]: ACTIVE, INACTIVE, LOCKED would be a good use-case for an ENUM

File: frontend/src/components/Users/UserInfo/UserInfo.jsx [hunk 2]
```diff
@@ -1,55 +1,119 @@
...
+        { id: 3, name: "LOCKED" }
+    ];
+
+    const { data: divisions, error: errorDivisions, isLoading: isLoadingDivisions } = useGetDivisionsQuery();
+    const { data: roles, error: errorRoles, isLoading: isLoadingRoles } = useGetRolesQuery();
+
+    useEffect(() => {
```
Comment (fpigeonjr) [inline frontend/src/components/Users/UserInfo/UserInfo.jsx:20]: 🔥nice useEffect

File: frontend/src/components/Users/UserInfo/UserInfo.module.css [hunk 3]
```diff
@@ -1,4 +1,6 @@
 /*Placeholder CSS to stub out space on the page.*/
-.container {
-    height: 1000px;
+
+.centeredItem {
+    align-items: center;
+    display: flex;
```
Comment (fpigeonjr) [inline frontend/src/components/Users/UserInfo/UserInfo.module.css:5]: may be able to use USWDS classes here like: "display-flex, flex-align-center"?

**Review by fpigeonjr — CHANGES_REQUESTED** [review]
⭐ User-Admin is much improved. Have a few observations that don't pass the how-to-test section.

**Comment by johndeange** [comment]
I apologize @fpigeonjr , I messed up the "How to Test" section.  You are correct that the test user with the `USER_ADMIN` role is the `Budget Team` user.  I updated the testing instructions.

**Review by maiyerlee — APPROVED** [review]
I followed the steps in How To Test and the changes checked out.

**Review by fpigeonjr — APPROVED** [review]

File: frontend/src/components/Users/UserInfo/UserInfo.constants.js [hunk 4]
```diff
@@ -0,0 +1,13 @@
+/**
+ * Enum for User status.
+ *
+ * @enum {string}
+ * @property {string} ACTIVE - Represents an active status for a user.
+ * @property {string} INACTIVE - Represents an inactive status for a user.
+ * @property {string} LOCKED - Represents a locked status for a user.
+ */
+export const USER_STATUS = {
```
Comment (fpigeonjr) [inline frontend/src/components/Users/UserInfo/UserInfo.constants.js:9]: 🤩

**Pull request merged**